lib/StripeClient.php

```diff
@@ -0,0 +1,146 @@
+     * @param string|null $filesBase The base URL for Stripe's Files API. Defaults to {@link DEFAULT_FILES_BASE}.
+     */
+    public function __construct(
+        $apiKey,
+        $clientId = null,
+        $apiBase = null,
+        $connectBase = null,
+        $filesBase = null
+    ) {
+        if (!is_null($apiKey) && ($apiKey === '')) {
+            $msg = "API key cannot be the empty string.";
+            throw new \Stripe\Exception\InvalidArgumentException($msg);
+        }
+        if (!is_null($apiKey) && (preg_match('/\s/', $apiKey))) {
+            $msg = "API key cannot contain whitespace.";
+            throw new \Stripe\Exception\InvalidArgumentException($msg);
+        }
+
+        $this->apiKey = $apiKey;
+        $this->clientId = $clientId;
+
+        $this->apiBase = $apiBase ?: self::DEFAULT_API_BASE;
+        $this->connectBase = $connectBase ?: self::DEFAULT_CONNECT_BASE;
+        $this->filesBase = $filesBase ?: self::DEFAULT_FILES_BASE;
+    }
+
+    /**
+     * Gets the API key used by the client to send requests.
+     *
+     * @return string|null The API key used by the client to send requests.
+     */
+    public function getApiKey()
+    {
+        return $this->apiKey;
+    }
+
+    /**
+     * Gets the client ID used by the client in OAuth requests.
+     *
+     * @return string|null The client ID used by the client in OAuth requests.
+     */
+    public function getClientId()
+    {
+        return $this->clientId;
+    }
+
+    /**
+     * Gets the base URL for Stripe's API.
+     *
+     * @return string The base URL for Stripe's API.
+     */
+    public function getApiBase()
+    {
+        return $this->apiBase;
+    }
+
+    /**
+     * Gets the base URL for Stripe's OAuth API.
+     *
+     * @return string The base URL for Stripe's OAuth API.
+     */
+    public function getConnectBase()
+    {
+        return $this->connectBase;
+    }
+
+    /**
+     * Gets the base URL for Stripe's Files API.
+     *
+     * @return string The base URL for Stripe's Files API.
+     */
+    public function getFilesBase()
+    {
+        return $this->filesBase;
+    }
+
+    /**
+     * Sends a request to Stripe's API.
+     *
+     * @param string $method The HTTP method.
+     * @param string $path The path of the request.
+     * @param array $params The parameters of the request.
+     * @param array|\Stripe\RequestOptions $opts The special modifiers of the request.
+     * @return \Stripe\StripeObject The object returned by Stripe's API.
+     */
+    public function request($method, $path, $params, $opts)
+    {
+        $opts = \Stripe\Util\RequestOptions::parse($opts);
+        $baseUrl = $opts->apiBase ?: $this->getApiBase();
+        $requestor = new \Stripe\ApiRequestor($this->apiKeyForRequest($opts), $baseUrl);
+        list($response, $opts->apiKey) = $requestor->request($method, $path, $params, $opts->headers);
```

Comment:
it's weird to me that we override $opts to store the API key for the next line. You said it's to handle `save` on the object but we should consider banning save if you use a Client.

Reply:
👍 Will try to think of a way to avoid storing the API key and freeze the resulting object when it's instantiated by `StripeClient`.